### DISCUSSION TITLE
Get all their command line arguments

### DISCUSSION THREAD

**Discussion by GalacticBard08** [pytest-dev/pytest#9993] [Q&A]
I'm asking some arguments:

    def pytest_addoption(parser):
       parser.addoption('--t')
       parser.addoption('--i', default='/home')
       parser.addoption('--o', default='/home')
       parser.addoption('--f')
       parser.addoption('--s')
       parser.addoption('--p')

There can be quite a lot of these arguments, how can I get the values of these arguments specified at startup in the command line all at once? Without creating a separate function to get each argument?

From `request.congig` I can't pull out the naming of all the arguments
I can do this using `nameGroup.options[0].dest` through the cycle

But this code does not look like a solution to my question. Are there any other options to get all the arguments (and values, if possible) at once?

**Comment by GalacticBard08** [comment]
I started solving this question on [stackoverflow](https://stackoverflow.com/questions/72355008/how-do-i-get-all-the-arguments-specified-on-the-command-line-in-a-list-pytest/72377735#72377735), but I didn't get any answer there

**Comment by nicoddemus** [comment]
Hi @forestnew,

Currently there's no API to obtain a list of all options (and values) defined in `pytest.Config`.
